crates/primitives/starknet/src/execution/felt252_wrapper.rs

```diff
@@ -0,0 +1,373 @@
+//! # Felt252 - FieldElement wrapper.
+//!
+//! Starknet base type is a [`FieldElement`] from starknet-ff crate.
+//! Substrate primitives are passed back and forth between client
+//! and runtime using SCALE encoding: https://docs.substrate.io/reference/scale-codec/.
+//!
+//! The [`Felt252Wrapper`] implements the traits for SCALE encoding, and wrap
+//! the [`FieldElement`] type from starknet-ff.
+
+use core::ops::Deref;
+
+use cairo_vm::felt::Felt252;
+use scale_codec::{Decode, Encode, EncodeLike, Error, Input, MaxEncodedLen, Output};
+use scale_info::build::Fields;
+use scale_info::{Path, Type, TypeInfo};
+use sp_core::{H256, U256};
+use starknet_ff::{FieldElement, FromByteSliceError, FromStrError};
+
+///
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct Felt252Wrapper(pub FieldElement);
```

Comment:
can't we use [`DerefMut`](https://doc.rust-lang.org/stable/std/ops/trait.DerefMut.html) to avoid all the re-implementations below?

Reply:
Tried and doesn't work unfortunately, maybe I'm doing it wrong but seems that we still need these re-implementations.

```rust
impl DerefMut for Felt252Wrapper {
    fn deref_mut(&mut self) -> &mut Self::Target {
        return &mut self.0;
    }
}
```

Reply:
@tdelabro an idea?